src/apexguru/apex-guru-service.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as vscode from 'vscode';
+import * as fspromises from 'fs/promises';
+import { CoreExtensionService } from '../lib/core-extension-service';
+import { ApexGuruAuthResponse, ApexGuruInitialResponse } from '../types';
+import * as Constants from '../lib/constants';
+
+export async function isApexGuruEnabledInOrg(outputChannel: vscode.LogOutputChannel): Promise<boolean> {
+    try {
+		const connection = await CoreExtensionService.getConnection();
+		const response:ApexGuruAuthResponse = await connection.request({
+			method: 'GET',
+			url: Constants.APEX_GURU_AUTH_ENDPOINT,
+			body: ''
+		});
+		return response.status == 'Success';
+	} catch(e) {
+		// This could throw an error for a variety of reasons. The API endpoint has not been deployed to the instance, org has no perms, timeouts etc,.
+		// In all of these scenarios, we return false.
+		const errMsg = e instanceof Error ? e.message : e as string;
+		outputChannel.error('***ApexGuru perm check failed with error:***' + errMsg);
+		outputChannel.show();
+		return false;
+	}
+}
+
+export async function runApexGuruOnFile(selection: vscode.Uri, outputChannel: vscode.LogOutputChannel) {
+	try {
+        const requestId = await initiateApexGuruRequest(selection, outputChannel);
+		// TODO: Logging the request Id for easy QA. Future stories will use this requestId to poll and retrieve the Apex Guru report.
+        outputChannel.appendLine('***Apex Guru request Id:***' + requestId);
+    } catch (e) {
+        const errMsg = e instanceof Error ? e.message : e as string;
+        outputChannel.appendLine('***Apex Guru initiate request failed***');
+        outputChannel.appendLine(errMsg);
+    }
+}
+
+export async function initiateApexGuruRequest(selection: vscode.Uri, outputChannel: vscode.LogOutputChannel) {
+	const fileContent = await fileSystem.readFile(selection.fsPath);
+	const base64EncodedContent = Buffer.from(fileContent).toString('base64');
+	const connection = await CoreExtensionService.getConnection();
+	const response: ApexGuruInitialResponse = await connection.request({
+		method: 'POST',
+		url: Constants.APEX_GURU_REQUEST,
+		body: JSON.stringify({
+			classContent: base64EncodedContent
+		})
+	});
+
+	if (response.status != 'new' && response.status != 'success') {
+		outputChannel.warn('***Apex Guru returned unexpected response:***' + response.status);
+		return '';
```

Comment:
I'm guessing we'll be adding in polling soon (in a future PR) in order to wait for the job to finish so we can get the results back and process the output. So my question is, is returning an empty string really what we want when the initiate request doesn't succeed. I would think we would just error and not return in order to prevent polling from even starting.

Reply:
A lot of this is TBD and so I just returned an empty string. Sure, I went ahead and made it return an error instead.